src/main/java/mate/academy/spring/validator/PasswordValidator.java

```diff
@@ -0,0 +1,27 @@
+package mate.academy.spring.validator;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import javax.validation.ConstraintValidator;
+import javax.validation.ConstraintValidatorContext;
+import mate.academy.spring.dto.request.UserRequestDto;
+
+public class PasswordValidator implements ConstraintValidator<Password, UserRequestDto> {
+    private static final String PASSWORD_PATTERN =
+            "/^(?=.*[0-9])(?=.*[a-z])(?=.*[A-Z])(?=.*\\W)(?!.* ).{8,16}$/;";
+    private static final int MIN_LENGTH = 8;
+
+    @Override
+    public boolean isValid(UserRequestDto userRequestDto,
+                           ConstraintValidatorContext constraintValidatorContext) {
+        String password = userRequestDto.getPassword();
+        String repeatPassword = userRequestDto.getRepeatPassword();
+        if (password == null || !password.equals(repeatPassword)
+                || (password.length() < MIN_LENGTH)) {
+            return false;
+        }
+        Pattern pattern = Pattern.compile(PASSWORD_PATTERN);
+        Matcher matcher = pattern.matcher(password);
+        return matcher.matches();
```

Comment:
```suggestion
        return pattern.matcher(password).matches();
```
You can simplify this